src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessor.java

```diff
@@ -0,0 +1,204 @@
+     * @param searchPhaseResult {@link SearchPhaseResults} DTO that has query search results. Results will be mutated as part of this method execution
+     * @param searchPhaseContext {@link SearchContext}
+     * @param <Result>
+     */
+    @Override
+    public <Result extends SearchPhaseResult> void process(
+        final SearchPhaseResults<Result> searchPhaseResult,
+        final SearchPhaseContext searchPhaseContext
+    ) {
+        if (searchPhaseResult instanceof QueryPhaseResultConsumer) {
+            QueryPhaseResultConsumer queryPhaseResultConsumer = (QueryPhaseResultConsumer) searchPhaseResult;
+            Optional<SearchPhaseResult> maybeResult = queryPhaseResultConsumer.getAtomicArray()
+                .asList()
+                .stream()
+                .filter(Objects::nonNull)
+                .findFirst();
+            if (isNotHybridQuery(maybeResult)) {
+                return;
+            }
+
+            TopDocsAndMaxScore[] topDocsAndMaxScores = getCompoundQueryTopDocsForResult(searchPhaseResult);
+            CompoundTopDocs[] queryTopDocs = Arrays.stream(topDocsAndMaxScores)
+                .map(td -> td != null ? (CompoundTopDocs) td.topDocs : null)
+                .collect(Collectors.toList())
+                .toArray(CompoundTopDocs[]::new);
+
+            ScoreNormalizer scoreNormalizer = new ScoreNormalizer();
+            scoreNormalizer.normalizeScores(queryTopDocs, normalizationTechnique);
+
+            ScoreCombiner scoreCombinator = new ScoreCombiner();
+            scoreCombinator.combineScores(topDocsAndMaxScores, queryTopDocs, combinationTechnique);
+
+            updateOriginalQueryResults(searchPhaseResult, queryTopDocs);
+        }
+    }
+
+    @Override
+    public SearchPhaseName getBeforePhase() {
+        return SearchPhaseName.QUERY;
+    }
+
+    @Override
+    public SearchPhaseName getAfterPhase() {
+        return SearchPhaseName.FETCH;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public String getTag() {
+        return tag;
+    }
+
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public boolean isIgnoreFailure() {
+        return true;
+    }
+
+    protected void validateParameters(final String normalizationTechniqueName, final String combinationTechniqueName) {
+        if (Strings.isNullOrEmpty(normalizationTechniqueName)) {
+            throw new IllegalArgumentException("normalization technique cannot be empty");
+        }
+        if (Strings.isNullOrEmpty(combinationTechniqueName)) {
+            throw new IllegalArgumentException("combination technique cannot be empty");
+        }
+        if (!EnumUtils.isValidEnum(ScoreNormalizationTechnique.class, normalizationTechniqueName)) {
+            log.error(String.format(Locale.ROOT, "provided normalization technique [%s] is not supported", normalizationTechniqueName));
```

Comment:
I put customer provided values only to log message, and exception has only general info that "value is invalid". This is to minimize security risks of rendering user input back, as this may be a malicious content